wardenclyffe/main/views.py

```diff
@@ -550,21 +529,10 @@ def s3_batch_upload(request, collection_id):
             language=request.POST.get('language_' + idx, ""),
         )
 
-        source_file = v.make_source_file(key)
-        label = "uploaded source file (S3)"
-        source_file = File.objects.create(video=v, url="", cap=key,
-                                          location_type="s3",
-                                          filename=key,
-                                          label=label)
-
-        v_operations = [
-            v.make_pull_from_s3_and_extract_metadata_operation(
-                key=key, user=request.user),
-            v.make_create_elastic_transcoder_job_operation(
-                key=key, user=request.user)]
-        if v.collection.audio:
-            o = v.make_audio_encode_operation(source_file.id, request.user)
-            v_operations.append(o)
+        v.make_source_file(key)
+        v.make_uploaded_source_file(key)
+        v_operations = v.initial_operations(key, request.user,
```

Comment:
Yeah, I was naughty and didn't stick to pure refactoring there. As I put that one in, I noticed that the batch upload behavior still had the old (pre-full-S3) approach to audio encoding (I suspect no one has tried to do a batch upload of audio files recently, so no one encountered it), so I just switched it to the new approach while I was at it.